docs/envoy.md

```diff
@@ -0,0 +1,326 @@
+# Life of an OSM Request
```

Comment:
This should go into the osm-docs repo instead, along with the traffic management docs: https://github.com/openservicemesh/osm-docs/tree/main/content/docs/tasks_usage/traffic_management

Reply:
https://github.com/openservicemesh/osm-docs/pull/132